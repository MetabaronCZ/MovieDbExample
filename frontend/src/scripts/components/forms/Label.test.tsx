import { getByText } from '@testing-library/dom';
import { describe, expect, it } from '@jest/globals';

import { Label } from './Label';
import { TestComponentWrapper, waitForComponent } from 'test-utils/component';

describe('components/forms/Label', () => {
  it('should render properly', async () => {
    const { container } = await waitForComponent(
      <Label label="CONTENT" htmlFor="ID" />,
      { wrapper: TestComponentWrapper },
    );
    expect(container).toMatchSnapshot();

    const label = getByText(container, 'CONTENT');
    expect(label).toBeInTheDocument();
    expect(label.tagName.toLowerCase()).toEqual('label');
    expect(label).toHaveAttribute('for', 'ID');
  });

  it('should render required state', async () => {
    const { container } = await waitForComponent(
      <Label label="CONTENT" required />,
      { wrapper: TestComponentWrapper },
    );
    expect(container).toMatchSnapshot();

    const label = getByText(container, 'CONTENT');
    expect(label).toBeInTheDocument();

    const asterisk = label.querySelector('span');
    expect(asterisk).toBeInTheDocument();
  });

  it('should render compact variant', async () => {
    const { container } = await waitForComponent(
      <Label label="CONTENT" compact />,
      { wrapper: TestComponentWrapper },
    );
    expect(container).toMatchSnapshot();
  });
});
