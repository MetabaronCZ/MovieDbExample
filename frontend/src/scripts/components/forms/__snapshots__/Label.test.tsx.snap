// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/forms/Label should render properly 1`] = `
.c0 {
  font-family: Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #002d56;
  display: flex;
  flex-direction: row;
  gap: 4px;
}

<div>
  <label
    class="c0"
    for="ID"
  >
    CONTENT
  </label>
</div>
`;

exports[`components/forms/Label should render required state 1`] = `
.c0 {
  font-family: Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #002d56;
  display: flex;
  flex-direction: row;
  gap: 4px;
}

.c1 {
  color: #f00;
}

.c1::after {
  content: '*';
}

<div>
  <label
    class="c0"
  >
    <span
      class="c1"
    />
    CONTENT
  </label>
</div>
`;
